### ISSUE TITLE
3D, 2D, LCD Shaders not showing after update. 

### ISSUE THREAD

**Issue by sinistarZ** [dragoonDorise/EmuDeck#717]
After the latest update, all emulators that had 3D, 2D and LCD shaders no longer show anymore. 

I tried quick reset with no success, then I went back and switched them all off, then back on but they remain off

**Comment by Godsbane** [comment]
can you please validate that you have your games in the base folders? (no sub folders)

**Comment by sinistarZ** [comment]
All my games are in the base folders:
![image](https://user-images.githubusercontent.com/64511782/234989212-d61e8751-103f-4dae-8a65-1dbf30bfe0ca.png)

except for PSX and Saturn, for those 2 each game is a sub folder

**Comment by Godsbane** [comment]
Ok please upload the ~/emudeck/emudeck.log and 

Zip up this folder ~/.var/app/org.libretro.RetroArch/config/retroarch/config and upload it as well.

**Comment by Godsbane** [comment]
retroarch config wasn't changed from last build - so if something is wrong it's a regression.  Please also upload the log and configs.

**Comment by Godsbane** [comment]
Thanks for uploading this.

I'll focus on NES for now.

`video_shader_enable = "true"` is set properly
`#reference /app/share/libretro/shaders/shaders_slang/crt/crt-aperture.slangp` is in the slangp file.

Please ensure your game is in the folder `nes` and let me know if the shader is still not applied as expected.

**Comment by D-Fox** [comment]
`nes` is ok. I see loaded `crt-aperture.slangp` shader in RetroArch Shader options. But no shader is loader for `gb by default.

**Comment by Godsbane** [comment]
Ok i'll take a look there. which core please?

**Comment by D-Fox** [comment]
Gambatte

**Comment by Godsbane** [comment]
```
appending: video_shader_enable = "true" to /home/deck/.var/app/org.libretro.RetroArch/config/retroarch/config/Gambatte/gb.cfg
Deleting video_filter from /home/deck/.var/app/org.libretro.RetroArch/config/retroarch/config/Gambatte/gb.cfg
appending: video_smooth = "false" to /home/deck/.var/app/org.libretro.RetroArch/config/retroarch/config/Gambatte/gb.cfg
```

is set to: #reference "/app/share/libretro/shaders/shaders_glsl/handheld/zfast_lcd.glslp"

Could be a case of a shader name problem, will investigate since i am not familiar with this shader. @dragoonDorise thoughts?

**Comment by Godsbane** [comment]
lcd shaders should be all set. Reset RA config and then re-enable them (regardless of setting) in quick settings.  Let me know how it goes.

**Comment by sinistarZ** [comment]
After a full wipe and reinstall the issue persisted. Since opening this there were multiple updates to emudeck and it's configs. The lcd3x shader I wanted back has been removed for zfastlcd as default in the config. This issue is stale

**Comment by Godsbane** [comment]
It should be working without issue.

We've tested multiple times and the shader shows up correctly.

**Comment by sinistarZ** [comment]
Are you referring to lcd3x or zfastlcd?

**Comment by Godsbane** [comment]
I am referring to zfast, as there  there was an actual issue with it not applying. It should now apply after resetting the config in RetroArch. If you can validate this, that would be great. It is VERY similar to the lcd3x shader, except there's better pixel alignment.

The shader can be adjusted (by the user) by simply changing the preset after enabling it. The old shader (lcd3x) still exists, and can be accessed in the internal libretro/shaders folder, or you can go to Online Updater --> Shaders and have it put them in the .var folder for use as well.

I can walk you through this if you can validate that the shaders did in fact enable correctly.

it's worth noting i did also ask for the config files from your device, to validate that there are no overrides or anything blocking it. You can still feel free to upload those.

**Comment by sinistarZ** [comment]
Thank you for offering help, I never meant to come off that you hadn't otherwise. I didn't upload my config files because at the time of asking me for the config files I already had wiped my deck, then by the time I saw this I had already set a content directory override so I didn't think it would have been any help. If you still think that would be useful, I can still upload that.

On another note, is there somewhere I can read up on where the conversation for lcd3x being replaced by zfast? I believe you when you say it has better pixel alignment, but for me personally after spending a few days of switching between the 2 shaders, handheld and docked mode, in my case I really do feel lcd3x mimics the TFT LCD screen seen on the GBA better then zfast does. To my untrained and unexperienced-aside-from-gaming-eye, lcd3x appears pixel perfect in alignment with everything I've played so far where the art fits on the grid. I'm not here to argue this with more experienced then me developers, I genuinely just want to see proof for my self where there zfast has better pixel alignment over lcd3x, so I can make the decision for my self and move on
